app/controllers/user_credentials_controller.rb

```diff
@@ -14,6 +14,16 @@ def index
     end
   end
 
+  def import
+    file = params[:file]
+    UserCredential.import(file)
```

Comment:
Agreed.  Or in `UserCredential.import`, use `current_user.credentials.build(hash)`.